go/host/db/blocks.go

```diff
@@ -70,3 +97,18 @@ func (db *DB) readBlockHeader(r ethdb.KeyValueReader, hash gethcommon.Hash) (*ty
 	db.blockReads.Inc(1)
 	return header, nil
 }
+
+// Stores the tip batch header hash into the database.
+func (db *DB) writeTipBlockHash(w ethdb.KeyValueWriter, hash gethcommon.Hash) error {
```

Comment:
I guess the end-goal would be that obscuroscan has it's own block processor, we're it processes all the data it needs to syncronously display it, but in the meanwhile we're piggybacking on the obscuro node.

The L1 client isn't used in obscuroscan right ? And using it would be funky as we would get some data from the host and some data from the l1 client, potentially rendering a mismatch situation.

But, are you suggesting that 
```
func (s *ScanAPI) GetLatestBlockHeader() (*types.Header, error) {
	return s.host.DB().GetTipBlockHeader()
}
```
becomes something like
```
func (s *ScanAPI) GetLatestBlockHeader() (*types.Header, error) {
	return s.host.L1Client().GetLatestBlockHeader()
}
```

ie: instead of hitting the host db it would hit the host l1 client ?

Reply:
Yeah the second thing. But L1Repo() really so the API doesn't have to know where the data is coming from.

I'm going to refactor all the API's this week to use the services for data so for now I'd suggest just fudging in the L1Repo or the L1client however you can and I'll clean it up.